front/lib/api/assistant/generation.ts

```diff
@@ -112,9 +112,9 @@ export async function renderConversationForModel({
       });
     } else if (isContentFragmentType(m)) {
       messages.unshift({
-        role: "content_fragment" as const,
-        name: "inject_content_fragment",
-        content: `${m.title}\nCONTENT:\n${m.content}`,
+        role: "content_fragment",
+        name: "inject_content",
+        content: `TITLE: ${m.title}\nORIGIN: User provided content (${m.contentType})\nCONTENT:\n${m.content}`,
```

Comment:
This is not really true of slack I think we need to go a bit more specific here

how about varying the `name` based on the contentType

slack => inject_slack_thread_context
attachment => inject_user_attached_data

I think it would be enough and we could get away witht he ORIGIN thing as a first step. If we want to stick with ORIGIN I would call it TYPE

slack: TYPE: slack_thread_context
attachment: TYPE: user_attached_data

Reply:
Let's have assertNerver of course on these cases :pray:

Reply:
OK is it ok if I do it like this? in which case no need for assertnever i think?

Reply:
I do prefer a switch because it forces us to revisit these choice when we add a new one... whereas here we won't be alerted of the impact of adding a new one. But I can live with it :+1: